### DISCUSSION TITLE
Need help to minify Ubuntu image

### DISCUSSION THREAD

**Discussion by atharvashirdhankar** [slimtoolkit/slim#544] [Q&A]
Currently I'm trying to optimize ubuntu image which ubuntu as base image and gcc compiler and make installation setup on it.

When i try to minimize that image.
The slimmed completely erase the root directories inside the image.And the slimmed image becomes unusable.

I have gone through docs and found out exclude property for slimming image.

But dont know how to use it.
Can someone help me to optimize ubuntu image.
Thanks in advance 🙂.

**Comment by kcq** [comment]
Sounds like you have a generic image, which means you need to do extra work to optimize it. It's a lot easier when you have an application container image. When you have a generic image you don't know what it will need to do and it means you need to be more explicit about keeping/including the image components. Sounds like you want to use 'gcc' and 'make' in your image and you'll need all of their dependencies. I'd recommend starting with a more simple image.

There are several possible way to approach optimizing your image. For example, it'll be good to invoke gcc to compile something (a file you'd volume mount) when you run the `slim build` command. That way `slim` will be able to observe `gcc` and keep its components in the optimized image. You will likely need to use the `--include-*` flags (e.g., `--include-path`) to make sure you keep all header files `gcc` needs. The `--include-bin` flag might be useful too. This flag is like `--include-path` allowing you to keep the target binary file, but it also ensures that any shared libraries used by the target binary file are also kept in the optimized image. Try using the `--include-bin` flag on the `make` binary.

**Reply by atharvashirdhankar** [reply]
Thanks @kcq 😀 , I'll follow this instructions and slim build with the mentioned arguments.

**Comment by atharvashirdhankar** [comment]
What should be mentioned as value --include-bin value?

**Reply by atharvashirdhankar** [reply]
container contains bin directory in root of the container file structure and another inside usr/local/bin

What should I mentioned as the value?
